processing/src/main/java/io/druid/segment/data/IndexedDoubles.java

```diff
@@ -76,12 +84,13 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
         inspector.visit("indexed", IndexedDoubles.this);
         inspector.visit("offset", offset);
       }
-    };
+    }
+    return new HistoricalFloatColumnSelector();
   }
 
   default LongColumnSelector makeLongColumnSelector(ReadableOffset offset)
   {
-    return new LongColumnSelector()
+    class HistoricalLongColumnSelector implements LongColumnSelector, HistoricalColumnSelector
```

Comment:
I'm not familiar with java semantics for this part. Will this create a new class in a way different than an anonymous class since the class def is inline?

Reply:
https://docs.oracle.com/javase/tutorial/java/javaOO/anonymousclasses.html seems to indicate they should behave the same way